### PR TITLE
Document that component data is readonly

### PR DESCRIPTION
**Pull request by brianpeiris**
**Description:**
It's not obvious that component data is meant to be readonly. This can lead to confusion if a developer modifies data directly and it is later clobbered by a-frame.

**Changes proposed:**
- Add a note to the docs to clarify that component data is readonly and that `setAttribute` should be used instead.

### CODE REVIEW

File: docs/core/component.md [hunk 1]
```diff
@@ -244,12 +244,12 @@ the data to modify the entity. The handlers will usually interact with the
...
-| this.el.sceneEl | Reference to the [scene][scene] as an HTML element.                                                                                           |
-| this.id         | If the component can have [multiple instances][multiple], the ID of the individual instance of the component (e.g., `foo` from `sound__foo`). |
+| Property        | Description                                                                                                                                                                                                                                         |
+|-----------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| this.data       | Parsed component properties computed from the schema default values, mixins, and the entity's attributes. <br/>__Note:__ The data property should be considered readonly. If a component wants to update its own data, it should use *setAttribute*.|
```
Comment (dmarcos) [inline docs/core/component.md]: Should we remove it all together from the docs to avoid temptations? 😄

Comment (brianpeiris) [reply docs/core/component.md]: well, component authors need to know about the existence of `this.data`. Ideally we could solve this in code by using getters and setters and write-disabled properties but that's probably more work than it's worth.

Comment (dmarcos) [reply docs/core/component.md]: I was thinking that maybe the mention in the different methods documentation could be enough: https://aframe.io/docs/0.7.0/core/component.html#update-olddata
But yeah you also need to be aware of it for any other component method.

Comment (dmarcos) [reply docs/core/component.md]: What about:  
**Important:** Do not modify the data attribute directly. It is updated internally by A-Frame. To modify a component use *[setAttribute](https://aframe.io/docs/0.7.0/introduction/javascript-events-dom-apis.html#updating-a-component-with-setattribute)*

Comment (brianpeiris) [reply docs/core/component.md]: 👍 sounds good to me. I've pushed the change.

**Comment by dmarcos** [comment]
Sweet! Thanks 👍

**Comment by dmarcos** [comment]
Doc tests failing. It complains about the link to `setAttribute`

**Comment by dmarcos** [comment]
Thanks 🎉

**Pull request merged**